src/openms/source/ANALYSIS/OPENSWATH/PeakIntegrator.cpp

```diff
@@ -109,18 +109,19 @@ namespace OpenMS
     }
     else if (getIntegrationType() == "simpson")
     {
-      if (n_points < 3 || !(n_points % 2))
+      if (n_points < 3)
       {
-        LOG_DEBUG << std::endl << "Error in integratePeak: number of points must be >=3 and odd for Simpson's rule" << std::endl;
+        LOG_DEBUG << std::endl << "Error in integratePeak: number of points must be >=3 for Simpson's rule" << std::endl;
         return;
       }
-      for (auto it=chromatogram.RTBegin(left); it<chromatogram.RTEnd(right)-2; it=it+2)
+      const size_t last_pos = n_points % 2 ? 2 : 1;
+      for (auto it=chromatogram.RTBegin(left); it<chromatogram.RTEnd(right)-last_pos; it=it+2)
```

Comment:
It looks like `it<chromatogram.RTEnd(right)-last_pos` clips two or one point off of the end of the right most RT.  If this is the case, this would result in an underestimation of the peak area.  If there is an odd number of points, I would think there would be no need to subtract two points off of the end of the peak, correct?  For the case of an odd number of points, you could either subtract one point off of the end or add one point to the end.  Preferable, I would think it would be more accurate to calculate the area by first subtracting one point off, then adding one point to the end, and taking the average.  Is this what you are doing below?

Reply:
I did things fast, forgetting about the solution you had proposed. Let me redo this. Sorry!